spyder/plugins/explorer/widgets.py

```diff
@@ -851,7 +875,182 @@ def create_func(fname):
+                else:
+                    selected_item = fnames[0]
+            if osp.isfile(selected_item):
+                parrent_path = osp.dirname(selected_item)
+            else:
+                parrent_path = osp.normpath(selected_item)
+            cb_data = QApplication.clipboard().mimeData()
+            if cb_data.hasUrls():
+                urls = cb_data.urls()
+                for url in urls:
+                    source_name = url.toLocalFile()
+                    base_name = osp.basename(source_name)
+                    if osp.isfile(source_name):
+                        try:
+                            while base_name in os.listdir(parrent_path):
+                                file_no_ext, file_ext = osp.splitext(base_name)
+                                end_number = re.search(r'\d+$', file_no_ext)
+                                if end_number:
+                                    new_number = int(end_number.group()) + 1
+                                else:
+                                    new_number = 1
+                                left_string = re.sub(r'\d+$', '', file_no_ext)
+                                left_string += str(new_number)
+                                base_name = left_string + file_ext
+                                destination = osp.join(parrent_path, base_name)
+                            else:
+                                destination = osp.join(parrent_path, base_name)
+                            shutil.copy(source_name, destination)
+                        except Exception as e:
+                            QMessageBox.critical(self, _('Error pasting file'),
+                                                 _("Unsupported copy operation"
+                                                   ". The error was:\n\n")
+                                                 + to_text_string(e))
+                    else:
+                        try:
+                            while base_name in os.listdir(parrent_path):
+                                end_number = re.search(r'\d+$', base_name)
+                                if end_number:
+                                    new_number = int(end_number.group()) + 1
+                                else:
+                                    new_number = 1
+                                left_string = re.sub(r'\d+$', '', base_name)
+                                base_name = left_string + str(new_number)
+                                destination = osp.join(parrent_path, base_name)
+                            else:
+                                destination = osp.join(parrent_path, base_name)
+                            if osp.realpath(destination).startswith(
+                                    osp.realpath(source_name) + os.sep):
+                                QMessageBox.critical(self,
+                                                     _('Recursive Copy'),

```

Comment:
Shall I un-capitalise `Copy` here? make it similar to the rest you suggested.

Reply:
Yes, please.